File: nodejs/fibonacci.js
/**
 * Calculates Fib(n) recursively.
 * CSC 349 sample submission, Winter '19
 */

function fib(n) {
  if (n < 0) {
    return -1;
  }
  else if (n == 0 || n == 1) {
    return n;
  }
  else {
    return fib(n - 1) + fib(n - 2);
  }
}

console.log(fib(parseInt(process.argv[2])))

File: kotlin/Fibonacci.kt
/**
 * Calculates Fib(n) recursively.
 * CSC 349 sample submission, Winter '19
 */

fun main(args: Array<String>) {
    fun fib(n: Int): Int {
        return when {
            n < 0     -> -1
            n in 0..1 -> n
            else      -> fib(n - 1) + fib(n - 2)
        }
    }

    println(fib(args[0].toInt()))
}
